src/MLJInterface.jl

```diff
@@ -406,7 +406,383 @@ MLJModelInterface.metadata_model(
     input=MLJModelInterface.Table(MLJModelInterface.Continuous),
     target=AbstractVector{MLJModelInterface.Continuous},
     weights=true,
-    descr="Microsoft LightGBM FFI wrapper: Regressor",
+    human_name="LightGBM regressor",
 )
 
+"""
+$(MLJModelInterface.doc_header(LGBMRegressor))
+
+LightGBM, short for light gradient-boosting machine, is a
+framework for gradient boosting based on decision tree algorithms and used for
+classification, regression and other machine learning tasks, with a focus on
+performance and scalability. This model in particular is used for various types of
+regression tasks.
+
+# Training data 
+
+In MLJ or MLJBase, bind an instance `model` to data with 
+
+  mach = machine(model, X, y) 
+
+Here:
+
+- `X` is any table of input features (eg, a `DataFrame`) whose columns are of
+  scitype `Continuous`; check the column scitypes with `schema(X)`; alternatively,
+  `X` is any `AbstractMatrix` with `Continuous` elements; check the scitype with
+  `scitype(X)`.
+- y is a vector of targets whose items are of scitype `Continuous`. Check the
+  scitype with `scitype(y)`.
+
+Train the machine using `fit!(mach, rows=...)`.
+
+# Operations
+
+- `predict(mach, Xnew)`: return predictions of the target given new features
+  `Xnew`, which should have the same scitype as `X` above.
+
+# Hyper-parameters
+
+- `boosting::String = "gbdt"`: Which boosting algorithm to use. One of:
+    - gbdt: traditional gradient boosting
+    - rf: random forest
+    - dart: dropout additive regression trees
+    - goss: gradient one side sampling
+- `num_iterations::Int = 10`: Number of iterations to run the boosting algorithm.
+- `learning_rate::Float64 = 0.1`: The update or shrinkage rate. In `dart`
+  boosting, also affects the normalization weights of dropped trees.
+- `num_leaves::Int = 31`: The maximum number of leaves in one tree.
```

Comment:
@ablaom some parameters have constraints which are not updated in these docs, e.g. `num_leaves` default is `31` but the constraints are `(1 < _ <= 131072)`. Would MLJ users benefit from having them updated where relevant?

Reply:
Sure, if there are hard constraints, then document them, thanks.

Reply:
updated